### PR TITLE
Fix test/target enter exit data/test target enter exit data classes.cpp

### PR DESCRIPTION
**Pull request by josemonsalve2**
This is to tackle issue #17 as well as issue #10 . 

This test splits the old target `enter exit data` c++ test into two different tests for simple classes and a complex class that uses hierarchy. This test passes in gcc and xlc, but fails in clang

### CODE REVIEW

File: tests/4.5/target_enter_exit_data/test_target_enter_exit_data_classes_complex.cpp [hunk 1]
```diff
@@ -0,0 +1,194 @@
+//===--test_targe_enter_exit_data_classes_complex.c - test inheritance class -===//
+// 
+// OpenMP API Version 4.5 Nov 2015
+// 
+//
+// This test uses target enter data and exit data to map multiple levels of 
+// inheritance in classes and templates. The target enter and exit data are used
+// in the constructor and destructor of all the levels of the hierarchy. 
+// It requires the use of helper_ variables since it
+// is not legal to use the "this" pointer (implicitely or explicitely) in 4.5.
```
Comment (jhdavis8) [inline tests/4.5/target_enter_exit_data/test_target_enter_exit_data_classes_complex.cpp]: No 'e' in implicitly/explicitly

Comment (josemonsalve2) [reply tests/4.5/target_enter_exit_data/test_target_enter_exit_data_classes_complex.cpp]: Thanks Josh!

**Comment by josemonsalve2** [comment]
Yes, it is pretty much it, plus I added a test on the #target exit data to guarantee that the data is moved back during the destructor.  this is actually what's failing in Clang.

**Comment by spophale** [comment]
What is the message on clang failure ?

**Comment by josemonsalve2** [comment]
No particular error message, other than the values being incorrect:
See detailed information here. 
    https://crpl.cis.udel.edu/ompvvsollve/result_report/results.html?result_report=4b8a27d7d

**Pull request merged**